### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by harshitajain06**
- Ran the program using the command line.
- Set up associations between classes and objects.

### CODE REVIEW

File: associations.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  end
+end
+
+class Student
+  attr_accessor :name, :classroom
+
+  def initialize(name)
+    @name = name
+  end
+end
```
Comment (christianbiring1) [inline associations.rb]: - [ ] Kindly, as requested in the project requirements on the second point [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md#:~:text=Create%20the%20has%2Dmany/belongs%2Dto%20relationship%20between%20Classroom%20and%20Student.%20The%20following%20should%20be%20implemented%3A), make sure the relation `has-many` `belongs to` relationship is implemented. in such as way that `when adding a student to a classroom it also sets the classroom for the student` and `when setting the classroom for a student it also adds it to the classrooms' students.`
- [ ] Also the `Student`class is supposed to inherit from the `Person`

Hint: YOu may consider saying something like ;
```rb
class Student < Person
  attr_reader :classroom
  .
  .
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
end

```

File: associations.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (christianbiring1) [inline associations.rb]: - [ ] Kindy make sure `You create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental` as said in the requirements. A method to add a rental should be implemented here in the book and you must make sure it has `person`, `date` and the book its self.

Hint: 
```rb
def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: associations.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  attr_accessor :book, :person, :date
+
+  def initialize(book, person, date)
+    @book = book
+    @person = person
+    @date = date
+    @book.rentals << self
+    @person.rentals << self
+  end
+end
```
Comment (christianbiring1) [inline associations.rb:39]: - [ ] Great job :+1: n making the rental belongs to a book and also belongs to a person :100:

File: associations.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+class Person
+  attr_accessor :name, :rentals
+
+  def initialize(name)
+    @name = name
+    @rentals = []
+  end
+end
```
Comment (christianbiring1) [inline associations.rb]: - [ ] Kinldy makes sure a person has many rentals.

```rb
def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @harshitajain06 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# STATUS: REQUEST CHANGES :recycle:  
almost


## Highlights
- [ ] All linters checks are green :heavy_check_mark: 
- [ ] Descriptive PR `title` and `summary` :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by algerina — APPROVED** [review]
Hi @harshitajain06 ,

Great job so far! 

May I highlight the following: 
- [x] You  did well implementing the previous  TSE's requirements 👏  ✔️ 
- [x] It's a good  thing that you were able to get your linter checks to pass ✔️ 


## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

**Pull request merged**